backend/main.py

```
@@ -170,9 +201,18 @@
     except docker.errors.BuildError as e:
         logging.error(f"Docker build failed: {e}")
         yield f"data: {json.dumps({'stage': 'execution', 'status': 'error', 'message': f'Docker build failed: {e}'})}\n\n"
+    except docker.errors.DockerException as e:
+        error_message = f"Docker error: {e}"
+        if "CreateFile" in str(e) or "file specified" in str(e):
+            error_message = "Docker connection lost during execution. Please ensure Docker Desktop is running and try again."
+        logging.error(f"Docker error during execution: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)